waitress/task.py

```diff
@@ -571,14 +812,19 @@ def get_environment(self):
 
         # the following environment variables are required by the WSGI spec
         environ['wsgi.version'] = (1, 0)
-        environ['wsgi.url_scheme'] = wsgi_url_scheme
-        environ['wsgi.errors'] = sys.stderr # apps should use the logging module
+
+        # May have already been set by the proxy
+
+        if 'wsgi.url_scheme' not in environ:
+            environ['wsgi.url_scheme'] = request.url_scheme
```

Comment:
Maybe this could be set before the `if remote_peer == server.adj.trusted_proxy` to avoid the check.